main.cpp

```diff
@@ -13,75 +13,486 @@ What you create in this project will be used as the basis of Project 5 in the co
+        float gasMileage = 26.2f;        
+        int year = 1985;
+        std::string manufacturer = "Toyota";
+        std::string model = "Corolla";
+
+        //3) a member function whose parameter has a default value.
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
+    };
+
+    //wash and wax car
+    void washAndWaxCar(Car car); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior(Car car);
+    
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
+};
+
+
+/*
+
+Thing 1) GYM
+5 properties:
+    1) Number of Exercise Machines (int)
+    2) Number of Weekly Customers (float)
+    3) Average time people spend at GYM (float)
+    4) Monthly cost to customer (float)
+    5) amount of profit (float)
+3 things it can do:
+    1)provide excersise equipment 
+    2)charge customers
+    3)provide classes
+ */
+
+ struct GYM
+ {
+    //Number of Exercise Machines 
+    int numExerciseMachines = 18;
+    //Number of Weekly Customers 
+    float numWeeklyCustomers = 103.2f; 
+    //Average time people spend at GYM in minutes
+    float averageTimeSpend = 40.4f;
+    //Monthly cost to customer 
+    float monthlyPrice = 14.99f;
+    //amount of profit 
+    float amountOfProfit = 1100.f; 
+    //Tour of the gym to customers
+    void giveTour();
+    //charge customers
+    float chargeCustomer (bool monthlyPrice = true);
+    //provide classes
+    void provideClasses ();
+ };
+
+/*
+Thing 2) Games Console
+5 properties: 
+    1) Amount of storage memory (float)
+    2) Amount of downloaded games (int)
+    3) Amount of power consumed per hour rms (float)
+    4) HDMI Input and output (y/n) (bool)
+    5) Amount of usb slots (int)
+3 things it can do:
+    1)Connect to the internet
+    2)Play games from a disk
+    3)Play Videos
+ */
+struct GamesConsole
+{
+    //Amount of storage memory 
+    float storageInSSD = 256.f;
+    //Amount of downloaded games 
+    int downloadedGames = 4;
+    //Amount of power consumed per hour rms 
+    float WattsPerHourRMS = 100;
+    //HDMI Input and output 
+    bool hasHDMI= true; 
+    //Amount of usb slots 
+    int usbSlots = 4;
+    //Connect to the internet
+    struct HarrysConsole
+    {
+       //digital games on harrys console
+       int digitalGames = 10;
+       //number of controller
+       int controller = 2;
+       //Brand-new yes
+       bool isBrandNew = false;
+       //brand
+       std::string brand = "Playstation";
+       //price
+       float price = 499.99f;
+
+       void swapConsoleWithHarry();
+       void TurnOnHarrysConsole ();
+       float askHarryPrice ();
+    };
```

Comment:
`HarrysConsole` is basically an instance of a `GamesConsole`, that would be a member variable inside an instance of a `Human` class named `Harry`.
`HarrysConsole` should not be a nested class, as it does not have a strong relation to the class that it is nested inside.
A better choice for nested class would be `Game`
This concept of designing class hierarchies and nesting classes is an important concept to understand and nail early on in your journey as a programmer.   It will affect your design decisions years from now, so it's important to get it right now, while the concept is being learned.
- [ ] revise